### ISSUE TITLE
Ivy compilation failed

### ISSUE THREAD

**Issue by hugodes** [angular/angular#31056]
# 🐞 bug report

### Affected Package

    "@angular/core": "8.1.0-next.2",

### Is this a regression?

    "@angular/core": "8.0.x",

### Description

I tried out Ivy as suggested [here](https://blog.angular.io/its-time-for-the-compatibility-opt-in-preview-of-ivy-38f3542a282f) when compiling, I get :  ERROR in EISDIR: illegal operation on a directory, read
## 🔬 Minimal Reproduction

The error doesn't give me an indication as to where in my code the error would come from ...

## 🔥 Exception or Error
<pre><code>

$ npm start

> front@0.0.0 start /Users/hugodes/src/core-front
> gulp sass && node --max_old_space_size=3072 node_modules/.bin/ng serve -c=local

[17:43:41] Using gulpfile ~/src/core-front/gulpfile.js
[17:43:41] Starting 'sass'...
[17:43:42] Finished 'sass' after 724 ms
  0% compiling
Compiling @angular/core : es2015 as esm2015

Compiling @angular/common : es2015 as esm2015

Compiling @angular/platform-browser : es2015 as esm2015

Compiling @angular/platform-browser-dynamic : es2015 as esm2015

Compiling @angular/cdk/bidi : es2015 as esm2015

Compiling @angular/cdk/coercion : es2015 as esm2015

Compiling @angular/cdk/platform : es2015 as esm2015

Compiling @angular/cdk/keycodes : es2015 as esm2015

Compiling @angular/cdk/observers : es2015 as esm2015

Compiling @angular/cdk/a11y : es2015 as esm2015

Compiling @angular/animations : es2015 as esm2015

Compiling @angular/animations/browser : es2015 as esm2015

Compiling @angular/platform-browser/animations : es2015 as esm2015

Compiling @angular/material/core : es2015 as esm2015

Compiling @angular/common/http : es2015 as esm2015

Compiling @angular/material-moment-adapter : es2015 as esm2015

Compiling @angular/forms : es2015 as esm2015

Compiling @angular/router : es2015 as esm2015

Compiling @ngrx/store : es2015 as esm2015

Compiling @auth0/angular-jwt : module as esm5

Compiling @angular/animations/browser/testing : es2015 as esm2015

Compiling @angular/cdk : es2015 as esm2015

Compiling @angular/cdk/collections : es2015 as esm2015

Compiling @angular/cdk/scrolling : es2015 as esm2015

Compiling @angular/cdk/drag-drop : es2015 as esm2015

Compiling @angular/common/http/testing : es2015 as esm2015

Compiling @angular/cdk/portal : es2015 as esm2015

Compiling @angular/cdk/overlay : es2015 as esm2015

Compiling @angular/material/form-field : es2015 as esm2015

Compiling @angular/material/autocomplete : es2015 as esm2015

Compiling @angular/material/badge : es2015 as esm2015

Compiling @angular/cdk/layout : es2015 as esm2015

Compiling @angular/material/bottom-sheet : es2015 as esm2015

Compiling @angular/material/button : es2015 as esm2015

Compiling @angular/material/button-toggle : es2015 as esm2015

Compiling @angular/material/card : es2015 as esm2015

Compiling @angular/material/checkbox : es2015 as esm2015

Compiling @angular/material/chips : es2015 as esm2015

Compiling @angular/material/dialog : es2015 as esm2015

Compiling @angular/cdk/text-field : es2015 as esm2015

Compiling @angular/material/input : es2015 as esm2015

Compiling @angular/material/datepicker : es2015 as esm2015

Compiling @angular/material/divider : es2015 as esm2015

Compiling @angular/cdk/accordion : es2015 as esm2015

Compiling @angular/material/expansion : es2015 as esm2015

Compiling @angular/material/grid-list : es2015 as esm2015

Compiling @angular/material/icon : es2015 as esm2015

Compiling @angular/material/list : es2015 as esm2015

Compiling @angular/material/menu : es2015 as esm2015

Compiling @angular/material/select : es2015 as esm2015

Compiling @angular/material/tooltip : es2015 as esm2015

Compiling @angular/material/paginator : es2015 as esm2015

Compiling @angular/material/progress-bar : es2015 as esm2015

Compiling @angular/material/progress-spinner : es2015 as esm2015

Compiling @angular/material/radio : es2015 as esm2015

Compiling @angular/material/sidenav : es2015 as esm2015

Compiling @angular/material/slide-toggle : es2015 as esm2015

Compiling @angular/material/slider : es2015 as esm2015

Compiling @angular/material/snack-bar : es2015 as esm2015

Compiling @angular/material/sort : es2015 as esm2015

Compiling @angular/cdk/stepper : es2015 as esm2015

Compiling @angular/material/stepper : es2015 as esm2015

Compiling @angular/cdk/table : es2015 as esm2015

Compiling @angular/material/table : es2015 as esm2015

Compiling @angular/material/tabs : es2015 as esm2015

Compiling @angular/material/toolbar : es2015 as esm2015

Compiling @angular/cdk/tree : es2015 as esm2015

Compiling @angular/material/tree : es2015 as esm2015

Compiling @angular/material : es2015 as esm2015

Compiling @angular/core/testing : es2015 as esm2015

Compiling @angular/platform-browser/testing : es2015 as esm2015

Compiling @angular/compiler/testing : es2015 as esm2015

Compiling @angular/platform-browser-dynamic/testing : es2015 as esm2015

Compiling @angular/platform-server : es2015 as esm2015

Compiling @angular/platform-server/testing : es2015 as esm2015

Compiling @angular/common/testing : es2015 as esm2015

Compiling @angular/router/testing : es2015 as esm2015

Compiling @ngrx/effects : es2015 as esm2015

Compiling @ngrx/effects/testing : es2015 as esm2015

Compiling @ngrx/store/testing : es2015 as esm2015

Compiling @ngrx/store-devtools : es2015 as esm2015
                                                                                              
Date: 2019-06-14T15:45:17.206Z
Hash: 6b9902ab85a9ec6f706a
Time: 89574ms
chunk {main} main.js, main.js.map (main) 1.89 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 675 bytes [initial] [rendered]
chunk {polyfills-es5} polyfills-es5.js, polyfills-es5.js.map (polyfills-es5) 434 kB [initial] [rendered]
chunk {runtime} runtime.js, runtime.js.map (runtime) 6.08 kB [entry] [rendered]
chunk {styles} styles.js, styles.js.map (styles) 1.33 MB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 322 kB [initial] [rendered]

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

ERROR in EISDIR: illegal operation on a directory, read
** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
ℹ ｢wdm｣: Failed to compile.
ℹ ｢wdm｣: Compiling...
                                                                                              
Date: 2019-06-14T15:45:22.185Z - Hash: c5d40477487b59bbbed2 - Time: 4662ms
6 unchanged chunks

ERROR in EISDIR: illegal operation on a directory, read
ℹ ｢wdm｣: Failed to compile.
/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/transform/src/compilation.js:228
                    finally { if (e_2) throw e_2.error; }
                                       ^

Error: Cannot combine @Input decorators with query decorators
    at /Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/annotations/src/directive.js:350:23
    at Array.map (<anonymous>)
    at queriesFromFields (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/annotations/src/directive.js:346:23)
    at Object.extractDirectiveMetadata (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/annotations/src/directive.js:125:35)
    at ComponentDecoratorHandler.analyze (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/annotations/src/component.js:133:47)
    at analyze (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/transform/src/compilation.js:172:60)
    at _loop_1 (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/transform/src/compilation.js:212:25)
    at analyzeClass (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/transform/src/compilation.js:220:25)
    at visit (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/transform/src/compilation.js:234:21)
    at visitNodes (/Users/hugodes/src/core-front/node_modules/typescript/lib/typescript.js:16514:30)
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 8.0.2
Node: 10.16.0
OS: darwin x64
Angular: 8.1.0-next.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.800.2
@angular-devkit/build-angular      0.800.2
@angular-devkit/build-optimizer    0.800.2
@angular-devkit/build-webpack      0.800.2
@angular-devkit/core               8.0.2
@angular-devkit/schematics         8.0.2
@angular/cdk                       8.0.1
@angular/cli                       8.0.2
@angular/material                  8.0.1
@angular/material-moment-adapter   8.0.1
@ngtools/webpack                   8.0.2
@schematics/angular                8.0.2
@schematics/update                 0.800.2
rxjs                               6.5.2
typescript                         3.4.5
webpack                            4.30.0
</code></pre>

**Anything else relevant?**

I aslo followed the guide here: https://angular.io/guide/ivy

**Comment by AndrewKushnir** [comment]
Hi @hugodes,

Based on the error message, it looks like `@Input()` and `@ContentChild` decorators are used on the same field in a Component class. This is not supported in Ivy and as a possible solution, you can assign decorators to different class fields and use the value from the second field if the first one is not defined.

Thank you.

**Comment by hugodes** [comment]
Hi @AndrewKushnir, thanks for your reply.

After searching through the whole codebase, @ContentChild isn't used once, third party modules included.
I would gladly make a repo reproducing the problem, but the error trail doesn't give me any indication (at least I'm not seeing it) on where in the code the error comes from.

If interested I could send you a copy of our angular app privately.

**Comment by jpleclerc** [comment]
I have the same error in next.2.

"ERROR in EISDIR: illegal operation on a directory, read"

**Comment by AndrewKushnir** [comment]
@hugodes I just checked the code again and the error message is coming from [this condition](https://github.com/angular/angular/blob/master/packages/compiler-cli/src/ngtsc/annotations/src/directive.ts#L440): any query decorator (i.e. `ContentChild(ren)` and `ViewChild(ren)` ones, not only the `@ContentChild`) can not be combined with `@Input` decorator.

Could you please check the codebase again to see if there is a combined usage of `@Input` and query decorators?

Thank you.

**Comment by jpleclerc** [comment]
I built with verbose and prod and got a better error message:

`ERROR in EISDIR: illegal operation on a directory, read
Child mini-css-extract-plugin node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/raw-css-loader.js!node_modules/postcss-loader/src/index.js??extracted!node_modules/sass-loader/lib/loader.js??ref--13-3!src/styles.scss:
                   Asset     Size  Chunks  Chunk Names
    3rdpartylicenses.txt  1 bytes          
    Entrypoint mini-css-extract-plugin = *
    chunk    {0} * (mini-css-extract-plugin) 78.4 KiB [entry] [rendered]
     [/c4r] ./node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/raw-css-loader.js!./node_modules/postcss-loader/src??extracted!./node_modules/sass-loader/lib/loader.js??ref--13-3!./src/styles.scss 78.4 KiB {0} [built]
         ModuleConcatenation bailout: Module is not an ECMAScript module
         single entry !!C:\Users\jpleclerc\Desktop\portal\portal\Innova.Client\node_modules\@angular-devkit\build-angular\src\angular-cli-files\plugins\raw-css-loader.js!C:\Users\jpleclerc\Desktop\portal\portal\Innova.Client\node_modules\postcss-loader\src\index.js??extracted!C:\Users\jpleclerc\Desktop\portal\portal\Innova.Client\node_modules\sass-loader\lib\loader.js??ref--13-3!C:\Users\jpleclerc\Desktop\portal\portal\Innova.Client\src\styles.scss  mini-css-extract-plugin
`

**Comment by hugodes** [comment]
Thanks for your response @AndrewKushnir.

I searched through the whole codebase, and couldn't find one combination of input decorator and query decorators.

It would help if the stack trace pinpointed where in the code this problem was happening. Is there anything I can do  ?

I would love to push a repo with the repro, but for this I'd need to know where the error comes from.

Sending you our repo privately is still an option just in case.

**Comment by AndrewKushnir** [comment]
@hugodes I've created a PR to make error message more verbose (to include filename and a pointer to a specific class field) - #31123. You can try to make a local modification similar to the one from the PR I mentioned and see if that gives you more details. Thank you.

**Comment by hugodes** [comment]
@AndrewKushnir that was fast and awesome thanks ! I'll try it out and give you some better  feedback

**Comment by hugodes** [comment]
Hi @AndrewKushnir I tried to make the changes but in the node_modules folder for the files you modified I don't have the .ts files, only .d.ts and .js files.
So I guess I have to download the angular source from GitHub, make the changes and compile. Then use that custom build to build my app.
Is there a guide on how I could go about doing that.

Sorry for the beginner question, but I've never worked with angular this way before.

**Comment by AndrewKushnir** [comment]
Hi @hugodes, you can follow the steps [in this instruction](https://github.com/angular/angular/blob/master/docs/DEVELOPER.md) to build Angular locally. You can also checkout the changes from my PR and build packages with these changes included (so you don't have to update the code manually). Once you build Angular packages, you can use them instead of the existing ones in `node_modules`. Thank you.

**Comment by hugodes** [comment]
Thanks a bunch @AndrewKushnir !!

So I compiled from source (using the branch you used to create #31123 ). I then replaced the contents of the @angular folder in node_modules with the content of package-dist from the angular build.

Now when running ng serve, I get this error:

> $ ng serve
Schema validation failed with the following errors:
  Data path ".builders['app-shell']" should have required property 'class'.
Error: Schema validation failed with the following errors:
  Data path ".builders['app-shell']" should have required property 'class'.
    at MergeMapSubscriber._registry.compile.pipe.operators_1.concatMap.validatorResult [as project] (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/@angular-devkit/core/src/workspace/workspace.js:215:42)
    at MergeMapSubscriber._tryNext (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/operators/mergeMap.js:69:27)
    at MergeMapSubscriber._next (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/operators/mergeMap.js:59:18)
    at MergeMapSubscriber.Subscriber.next (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/Subscriber.js:67:18)
    at MergeMapSubscriber.notifyNext (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/operators/mergeMap.js:92:26)
    at InnerSubscriber._next (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/InnerSubscriber.js:28:21)
    at InnerSubscriber.Subscriber.next (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/Subscriber.js:67:18)
    at MapSubscriber._next (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/operators/map.js:55:26)
    at MapSubscriber.Subscriber.next (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/Subscriber.js:67:18)
    at SwitchMapSubscriber.notifyNext (/Users/hugodes/.nvm/versions/node/v10.16.0/lib/node_modules/@angular/cli/node_modules/rxjs/internal/operators/switchMap.js:86:26)

I'm guessing this might be a problem with @angualr/cli ? Am I doing something wrong ?

**Comment by AndrewKushnir** [comment]
@hugodes, could you please try installing `8.1.0-next.3` version of Angular packages from NPM to see if the error message is the same?

Also as an option, you can look at the generated code in `compiler-cli` package once you've built Angular project (based on the PR #31123) and apply the necessary changes to the `compiler-cli` package in your `node_modules` folder.

Thank you.

**Comment by hugodes** [comment]
Thanks so much for your continued support on this issue @AndrewKushnir !

With 8.1.0-next.3 I get the same error as I did with 8.1.0-next.2.

So I recompiled angular with the latest changes from your branch.
Then I did a diff -r angular/dist/packages-dist/compiler-cli myapp/node_modules/@angualr/compiler-cli

I then applied the changes from node_modules/@angular/compiler-cli/src/ngtsc/annotations/src/directive.js 
For info this is what the file looks like now from lines 345 to 362:

```javascript
function queriesFromFields(fields, reflector, evaluator) {
        return fields.map(function (_a) {
            var member = _a.member, decorators = _a.decorators;
            var decorator = decorators[0];
            var node = member.node || decorator.node;

          // Throw in case of `@Input() @ContentChild('foo') foo: any`, which is not supported in Ivy
            if (member.decorators.some(function (v) { return v.name === 'Input'; })) {
              throw new diagnostics_1.FatalDiagnosticError(diagnostics_1.ErrorCode.DECORATOR_COLLISION, node, 'Cannot combine @Input decorators with query decorators');
            }
            if (decorators.length !== 1) {
              throw new diagnostics_1.FatalDiagnosticError(diagnostics_1.ErrorCode.DECORATOR_COLLISION, node, 'Cannot have multiple query decorators on the same class member');
            }
            else if (!isPropertyTypeMember(member)) {
              throw new diagnostics_1.FatalDiagnosticError(diagnostics_1.ErrorCode.DECORATOR_UNEXPECTED, node, 'Query decorator must go on a property-type member');
            }
            return extractQueryMetadata(decorator.node, decorator.name, decorator.args || [], member.name, reflector, evaluator);
        });
    }
```

I then got a different error, but I still don't know from where in the code the error is coming from:

```
Date: 2019-06-21T09:52:44.152Z
Hash: 6b9902ab85a9ec6f706a
Time: 92200ms
chunk {main} main.js, main.js.map (main) 1.89 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 675 bytes [initial] [rendered]
chunk {polyfills-es5} polyfills-es5.js, polyfills-es5.js.map (polyfills-es5) 434 kB [initial] [rendered]
chunk {runtime} runtime.js, runtime.js.map (runtime) 6.08 kB [entry] [rendered]
chunk {styles} styles.js, styles.js.map (styles) 1.33 MB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 322 kB [initial] [rendered]

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

WARNING in Unknown member type: "[key: string]: (data: any) => void;

ERROR in EISDIR: illegal operation on a directory, read
** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
ℹ ｢wdm｣: Failed to compile.
ℹ ｢wdm｣: Compiling...
                                                                                              
Date: 2019-06-21T09:52:49.641Z - Hash: c5d40477487b59bbbed2 - Time: 5171ms
6 unchanged chunks

ERROR in EISDIR: illegal operation on a directory, read
ℹ ｢wdm｣: Failed to compile.
/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/transform/src/compilation.js:228
                    finally { if (e_2) throw e_2.error; }
                                       ^

Error: Unsupported binary operator: EqualsEqualsEqualsToken
    at StaticInterpreter.visitBinaryExpression (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:469:23)
    at StaticInterpreter.visitExpression (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:105:31)
    at StaticInterpreter.visitConditionalExpression (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:441:34)
    at StaticInterpreter.visitExpression (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:99:31)
    at StaticInterpreter.visitVariableDeclaration (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:253:29)
    at StaticInterpreter.visitDeclaration (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:232:29)
    at StaticInterpreter.visitIdentifier (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:210:31)
    at StaticInterpreter.visitExpression (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:90:31)
    at StaticInterpreter.visitObjectLiteralExpression (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:156:42)
    at StaticInterpreter.visitExpression (/Users/hugodes/src/core-front/node_modules/@angular/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.js:87:31)
```

TLDR: the error changed from `Cannot combine @Input decorators with query decorators` to `Unsupported binary operator: EqualsEqualsEqualsToken`

**Comment by AndrewKushnir** [comment]
Hi @hugodes, this is a bit strange that you have different error message now, since the code that I've added should just produce better diagnostics info. Anyway, the problem that you see should be addressed by PR #31145. My proposal is to wait for the next Angular release, that will most likely include changes from the mentioned PRs, try again and see what the error message is. Thank you.

**Comment by hugodes** [comment]
Ok thanks @AndrewKushnir, I'll update you on this issue as soon as 8.1-next-4 comes out.
Have a nice day

**Comment by istiti** [comment]
@AndrewKushnir maybe it is related, can you please try this repro in a clean project (cli & ivy v8.1 beta 3) : 
1. `npm install ngx-virtual-scroller`
2. by example in app.module import `VirtualScrollerModule`  from `import {VirtualScrollerModule} from 'ngx-virtual-scroller';`
3. then run `ng s`

It throw me same poor/insignificant error 
![image](https://user-images.githubusercontent.com/2789203/60181108-cea87500-9821-11e9-80ce-28d5a2cf8043.png)

**Comment by petebacondarwin** [comment]
https://github.com/angular/angular/issues/31056#issuecomment-505861134 is fixed by https://github.com/angular/angular/pull/31289

**Comment by AndrewKushnir** [comment]
@hugodes just want to let you know that `8.1.0-rc.0` is now released and available on NPM. Could you please try to reproduce original problem and see if there is additional debug information available? Thank you.

**Comment by hugodes** [comment]
@AndrewKushnir Thanks for the update !

So I just compiled the code, and the error went away. I'm now left with:

```code
ERROR in EISDIR: illegal operation on a directory, read
```

Should I open a new issue for this ?

Here is the full logs from the compiler:

```code
$ ng build
  0% compiling
Compiling @angular/core : module as esm5

Compiling @angular/common : module as esm5

Compiling @angular/platform-browser : module as esm5

Compiling @angular/platform-browser-dynamic : module as esm5

Compiling @angular/cdk/bidi : module as esm5

Compiling @angular/cdk/coercion : module as esm5

Compiling @angular/cdk/platform : module as esm5

Compiling @angular/cdk/keycodes : module as esm5

Compiling @angular/cdk/observers : module as esm5

Compiling @angular/cdk/a11y : module as esm5

Compiling @angular/animations : module as esm5

Compiling @angular/animations/browser : module as esm5

Compiling @angular/platform-browser/animations : module as esm5

Compiling @angular/material/core : module as esm5

Compiling @angular/common/http : module as esm5

Compiling @angular/material-moment-adapter : module as esm5

Compiling @angular/forms : module as esm5

Compiling @angular/router : module as esm5

Compiling @ngrx/store : module as esm5

Compiling @auth0/angular-jwt : module as esm5

Compiling @angular/animations/browser/testing : module as esm5

Compiling @angular/cdk : module as esm5

Compiling @angular/cdk/collections : module as esm5

Compiling @angular/cdk/scrolling : module as esm5

Compiling @angular/cdk/drag-drop : module as esm5

Compiling @angular/common/http/testing : module as esm5

Compiling @angular/cdk/portal : module as esm5

Compiling @angular/cdk/overlay : module as esm5

Compiling @angular/material/form-field : module as esm5

Compiling @angular/material/autocomplete : module as esm5

Compiling @angular/material/badge : module as esm5

Compiling @angular/cdk/layout : module as esm5

Compiling @angular/material/bottom-sheet : module as esm5

Compiling @angular/material/button : module as esm5

Compiling @angular/material/button-toggle : module as esm5

Compiling @angular/material/card : module as esm5

Compiling @angular/material/checkbox : module as esm5

Compiling @angular/material/chips : module as esm5

Compiling @angular/material/dialog : module as esm5

Compiling @angular/cdk/text-field : module as esm5

Compiling @angular/material/input : module as esm5

Compiling @angular/material/datepicker : module as esm5

Compiling @angular/material/divider : module as esm5

Compiling @angular/cdk/accordion : module as esm5

Compiling @angular/material/expansion : module as esm5

Compiling @angular/material/grid-list : module as esm5

Compiling @angular/material/icon : module as esm5

Compiling @angular/material/list : module as esm5

Compiling @angular/material/menu : module as esm5

Compiling @angular/material/select : module as esm5

Compiling @angular/material/tooltip : module as esm5

Compiling @angular/material/paginator : module as esm5

Compiling @angular/material/progress-bar : module as esm5

Compiling @angular/material/progress-spinner : module as esm5

Compiling @angular/material/radio : module as esm5

Compiling @angular/material/sidenav : module as esm5

Compiling @angular/material/slide-toggle : module as esm5

Compiling @angular/material/slider : module as esm5

Compiling @angular/material/snack-bar : module as esm5

Compiling @angular/material/sort : module as esm5

Compiling @angular/cdk/stepper : module as esm5

Compiling @angular/material/stepper : module as esm5

Compiling @angular/cdk/table : module as esm5

Compiling @angular/material/table : module as esm5

Compiling @angular/material/tabs : module as esm5

Compiling @angular/material/toolbar : module as esm5

Compiling @angular/cdk/tree : module as esm5

Compiling @angular/material/tree : module as esm5

Compiling @angular/material : module as esm5

Compiling @angular/core/testing : module as esm5

Compiling @angular/platform-browser/testing : module as esm5

Compiling @angular/compiler/testing : module as esm5

Compiling @angular/platform-browser-dynamic/testing : module as esm5

Compiling @angular/platform-server : module as esm5

Compiling @angular/platform-server/testing : module as esm5

Compiling @angular/common/testing : module as esm5

Compiling @angular/router/testing : module as esm5

Compiling @ngrx/effects : module as esm5

Compiling @ngrx/effects/testing : module as esm5

Compiling @ngrx/store/testing : module as esm5

Compiling @ngrx/store-devtools : module as esm5
                                                                                              
Date: 2019-06-27T08:22:48.049Z
Hash: a32c7b87e4c9cad1359f
Time: 96678ms
chunk {main} main-es5.js, main-es5.js.map (main) 624 bytes [initial] [rendered]
chunk {polyfills} polyfills-es5.js, polyfills-es5.js.map (polyfills) 435 kB [initial] [rendered]
chunk {runtime} runtime-es5.js, runtime-es5.js.map (runtime) 6.08 kB [entry] [rendered]
chunk {styles} styles-es5.js, styles-es5.js.map (styles) 1.33 MB [initial] [rendered]

WARNING in Unknown member type: "[key: string]: (data: any) => void;

ERROR in EISDIR: illegal operation on a directory, read
```

**Comment by petebacondarwin** [comment]
@hugodes - that is a known regression - sorry. Here is the fix https://github.com/angular/angular/pull/31289

**Comment by hugodes** [comment]
Thanks @petebacondarwin !

@AndrewKushnir Hey Andrew, so I made the changes from #31289 and the error pointed to a `@ViewChild` decorator !

But that `@ViewChild` child wasn't decorated by an `@Input()`.

So I set off to make a repo reproducing this issue, but no matter how close I copied the code from the original component I wasn't able to reproduce it.

So I put my view child who was on line 10 on line 11 and an input that was on line 11 on line 10, just to see if the error would still point to the line of code containing the `@ViewChild`.

And the error disappeared. 🤯

I'm kind of lost here, I don't see how switching up two lines in the code can change anything ...

For info, I switched it back and the error re-appeared.

Does this ring any bells on your end ?

**Comment by istiti** [comment]
It looks to me 
In Before : your input seinflowsum is decorated by viewchild
In after : it isn’t

Can you try :
```
@ViewChild('myHalfDonutCanvas', { static: false }) myHalfDonutCanvas;
  @Input() SEInflowSum;
  @Input() SEOutflowSum;
```

**Comment by AndrewKushnir** [comment]
Thanks for the detailed reply @istiti 👍 

@hugodes as Elvis mentioned, the following code:

```
  @ViewChild('myHalfDonutCanvas', { static: false })
  @Input() SEInflowSum;
```
means that `@ViewChild` and `@Input` decorators are applied to the `SEInflowSum` field, thus Ivy compiler throws an error. Please try to apply a fix that Elvis proposed and let us know if everything works as expected.

Thank you.

**Comment by hugodes** [comment]
@istiti Thank you so much !!

Also thanks @AndrewKushnir & @kara for all your help, I never had to wait more than 24h to get a response !

I'm sorry the error ended up being in my code, but other users will have better errors as a result 🙂

I created a separate issue #31327 for the case I ran into while trying to reproduce the error. (Sometimes combining input and query decorators doesn't fail with Ivy).

Have a nice day !

**Comment by richardsengers** [comment]
> @hugodes - that is a known regression - sorry. Here is the fix #31289

This fix hasn't made it in RC.0 right?

@hugodes how did you apply changes from #31289 as you stated above?


I'm having the exact same problem 
```
ERROR in EISDIR: illegal operation on a directory, read
```

And I would like to know where it comes from. I'm developing a large app, so checking my code if I miss a ; after an Query is a bit of a hassle :-)

**Comment by istiti** [comment]
@richardsengers to clarify in js ";" is not mandatory (but recommended) I ask him to add ";" to separate decorators and see different error than `Cannot combine @Input decorators with query decorators`, because actually without `;` his code is correct it is called "combinated annnotation" but these two aren't comptabile together in angular (ivy) 

Your error is something different than "combinated decorators" and thanks to @petebacondarwin #31289  should fix your issue.

No #31289  is not in 8.1rc0 but it is easy to fix it manually in this file `...\node_modules\@angular\compiler-cli\src\ngtsc\file_system\src\compiler_host.js` (assuming you have 8.1.0-rc.0 installed locally)

Replace line:60 by 
`return this.fs.exists(absPath) && this.fs.stat(absPath).isFile();`


@richardsengers Can you confirm it works with this temp fix ? 

@hugodes Can you maybe update your issue title to something more related to your issue instead just "ivy compilation failed"
Something to like `ivyget  ERROR in EISDIR: illegal operation on a directory, read and Cannot combine @Input decorators with query decorators` this will help people find issue easily in github search

**Comment by richardsengers** [comment]
@istiti thank you very much for clarifying. I think indeed changing line 60 fixed it, since I've got my originial error from before 8.1.0 back :-)
https://github.com/angular/angular/issues/31160

But they are working on a fix for that one too

**Comment by hugodes** [comment]
@istiti changed the issue title 😀
